src/nvim/ex_cmds2.c

```diff
@@ -1847,19 +1852,21 @@ void ex_argdelete(exarg_T *eap)
     /* ":1,4argdel": Delete all arguments in the range. */
     if (eap->line2 > ARGCOUNT)
       eap->line2 = ARGCOUNT;
-    int n = eap->line2 - eap->line1 + 1;
```

Comment:
For now linenumber differences are guaranteed to fit in an `int`. But you're right it's probably cleaner to keep it `linenr_T` as long as reasonable and then cast to an `int` when necessary.